firebase implemented and refactor

BasilBasilio opened the pull request:
firebase implemented and refactor

BasilBasilio committed:
firebase implemented and refactor

plushdohn commented on src/.env:
Due cose

## 1. Hai commitato il .env

pericoloso mistake questo, committare il `.env`.

il file `.env` solitamente contiene tutte le informazioni sensibili del progetto, in questo caso tutti i dati sul tuo progetto firebase. in questo caso potrei farmi un altro progetto come il tuo e accedere al progetto firebase che ti sei creato perche mi hai leakato queste informazioni (c'e un piccolo asterisco su questa cosa che sto tralasciando, ma voglio che capisca la gravita di committare un .env in modo che non ti succeda mai in una repository con altre persone o pubblica)

e' convenzione solitamente fare un `.env.example` che contiene i nomi delle variabili che servono ma con un valore d'esempio, cosi chi clona la tua repository e ci lavora vede che servono questi secret e te li puo chiedere in privato. questo chiaramente non va gitignorato, perche chi clona la tua repo deve poterlo vedere.

```
VITE_APIKEY="my-firebase-api-key"
// etc.
```

essenzialmente si, aggiungi al tuo .gitignore il `.env` e crea un `.env.example` con queste variabili ma con valori d'esempio

## 2. Userei un prefisso specifico

Tutte le variabili che hai messo riguardano firebase, se ne avrei altre sara molto incasinato capire cosa e' di firebase e cosa no. direi di prefissare tutte queste con `VITE_FIREBASE_` quindi `VITE_FIREBASE_API_KEY` per esempio, `VITE_FIREBASE_STORAGEBUCKET` ecc.

chiaramente ricordati di cambiare anche il relativo utilizzo lato codice

plushdohn commented on tailwind.config.js:
```diff
@@ -5,6 +5,8 @@ export default {
     "./src/**/*.{html,js}",
     "./src/components/**/*.{html,js,tsx}",
     "./src/components/*.{html,js,tsx}",
+    "./src/pages/*.{html,js,tsx}",
```
nitpick: tutti questi glob puoi semplificarli in uno singolo `./src/**/*.tsx`. un glob piu specifico puo rendere leggerissimamente piu veloce il tempo di compilazione di tailwind, perche deve controllare un range meno ampio di file per trovare le classi utilizzate.

plushdohn reviewed (changes requested):
dedicherei un po' di tempo a settarti un linter e un formatter

tendenzialmente si usano prettier e eslint

prettier e' un formatter, praticamente quando salvi il codice te lo autoformatta in una maniera consistente, che si applichera quindi sia per te che per chi collabora nel progetto

eslint e' un linter, cioe un tool per la qualita del codice

ti segnala possibili problemi vari come variabili non utilizzate, pattern sconsigliati, mancanza di dipendenze negli effect, ecc.


se ti serve una zampa a configurarli posso darti una mano, ma potresti anche riuscire da solo se te la senti

BasilBasilio committed:
.env added to .gitignore
Added prettier added userAuthContext Created UserSignUp component and added to login page

plushdohn commented on src/context/userAuthContext.tsx:
```diff
@@ -0,0 +1,83 @@
+        console.log('The logged in user state is: ', user);
+        setUser(user);
+      }
+      return () => {
```
questo va fuori da questo scope, ma sempre dentro l'effect

plushdohn commented on src/context/userAuthContext.tsx:
```diff
@@ -0,0 +1,83 @@
+  };
+
+  return (
+    <userAuthContext.Provider value={value}>
```
e' convenzione che il contest sia in PascalCase, non in camelCase

plushdohn commented on src/context/userAuthContext.tsx:
questo context secondo me fa troppo

ha senso che ci sia un context per tenere traccia di quando l'utente e' loggato (tutta la roba dentro l'effect) pero non mi sembra sensato creare una sottilissima astrazione attorno ai metodi di firebase per signIn, signUp, logout ecc. e passarli nel context

questi metodi non sono parte del view layer o della UI, sono metodi di firebase che non dipendono da uno state (a differenza dello user che ti stai storando in uno state) quindi puoi semplicemente importareli direttamente da firebase dove ti servono e usarli (es. nella pagina di login)

se ti sembra piu ordinato puoi valutare di crearti un file a parte (tipo `auth.ts`) dove ti riesporti queste funzioncine, cosi per esempio ti eviti di dover importare `auth` da firebase ogni volta, e invece importi solo le funzioni. la cosa principale e' che non ha senso che siano passate dal context.

BasilBasilio committed:
userAuthContext refactor
